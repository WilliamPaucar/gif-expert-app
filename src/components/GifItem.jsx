import PropTypes from 'prop-types'
import React from 'react'

export const GifItem = ({title, url}) => {
  return (
    <div className='card'>
        <img className='card img ' src={url} alt={url} />
        <br />
        <h4 className='card p'>{title}</h4>

    </div>
  )
}

GifItem.propTypes={
title :PropTypes.string.isRequired,
url:PropTypes.string.isRequired


}