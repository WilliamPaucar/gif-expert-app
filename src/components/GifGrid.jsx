
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const {images,isLoading}= useFetchGifs(category);

  return (
    <>
      <h2>{category}</h2>
    {isLoading && (<h2>Cargando...</h2>)}
      <ol>
        <div className="card-grid">
        {

        images.map((image) =>(
            <GifItem key={image.id} {...image}/>

        ))
        
      }
      </div>

      </ol>
    </>
  );
};

GifGrid.propTypes = {
  category:PropTypes.string.isRequired,
}