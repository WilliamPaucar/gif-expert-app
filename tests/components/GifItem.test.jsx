import { GifItem } from "../../src/components/GifItem";
import {render, screen} from "@testing-library/react"
 
describe('Probando el componente <GifItem/> ', () => { 
      const title= 'William'
      const url= 'http://localhost/www.josecita.com'
    test("debe hacer match con el snapshot", () => {

     const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
   });

   test('debe mostrar la imagen conr URL y el ALT indicado', () => { 

      render (<GifItem title={title} url={url}/>);
      const {src,alt}=screen.getByRole('img')
      expect(src).toBe(url);
      expect(alt).toBe(alt);
   })
   test('debe mostrar el titulo en el componente', () => { 

      render (<GifItem title={title} url={url}/>);

      expect(screen.getByText(title)).toBeTruthy;

    })
}) 