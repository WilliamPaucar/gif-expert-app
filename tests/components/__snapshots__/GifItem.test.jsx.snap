// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifItem/>  debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h4
      class="card p"
    >
      William
    </h4>
    <img
      alt="http://localhost/www.josecita.com"
      class="card img "
      src="http://localhost/www.josecita.com"
    />
    <br />
  </div>
</div>
`;
